haas/api.py

```diff
@@ -263,6 +263,19 @@ def node_power_off(node):
     node.obm.power_off()
 
 
+@rest_call('PUT', '/node/<node>/boot_device', Schema({
+    'node': basestring, 'bootdev': basestring,
+}))
+def node_set_bootdev(node, bootdev):
+    auth_backend = get_auth_backend()
+    node = _must_find(model.Node, node)
+    if node.project is None:
+        auth_backend.require_admin()
+    else:
+        auth_backend.require_project_access(node.project)
+    node.obm.set_bootdev(bootdev)
```

Comment:
> I really like that, as it keeps invalid values out of the lower depths.

I know I suggested that, but now that I think of it, a function like `validate_bootdev` would still be a part of the driver. It's just splitting the function into 2 parts  (one to do the validation, the other to set the boot device).